addons/crm/models/crm_lead.py

```diff
@@ -595,7 +595,16 @@ def write(self, vals):
         if any(field in ['active', 'stage_id'] for field in vals):
             self._handle_won_lost(vals)
 
-        write_result = super(Lead, self).write(vals)
+        # stage change between two won stages: does not chang the date_closed
+        if stage_is_won:
+            still_won_leads = self.filtered(lambda r: r.stage_id and r.stage_id.is_won)
```

Comment:
Note that the filtered could be simplified, r.stage_id.is_won should be sufficient (if not stage this will return False)